src/qtism/data/storage/xml/XmlResultDocument.php

```diff
@@ -38,7 +40,30 @@ class XmlResultDocument extends XmlDocument
      */
     public function getSchemaLocation(): string
     {
-        return __DIR__ . '/schemes/qtiv2p1/imsqti_result_v2p1.xsd';
+        $versionNumber = Version::appendPatchVersion($this->getVersion());
+
+        if ($versionNumber === '2.1.0') {
+            $filename = __DIR__ . '/schemes/qtiv2p1/imsqti_result_v2p1.xsd';
+        } elseif ($versionNumber === '2.1.1') {
+            $filename = __DIR__ . '/schemes/qtiv2p1/imsqti_result_v2p1.xsd';
+        } elseif ($versionNumber === '2.2.0') {
+            $filename = __DIR__ . '/schemes/qtiv2p2/imsqti_result_v2p2.xsd';
+        } elseif ($versionNumber === '2.2.1') {
+            $filename = __DIR__ . '/schemes/qtiv2p2/imsqti_result_v2p2.xsd';
+        } elseif ($versionNumber === '2.2.2') {
+            $filename = __DIR__ . '/schemes/qtiv2p2/imsqti_result_v2p2.xsd';
+        } else {
+            $knownVersions = ['2.1.0', '2.1.1', '2.2.0', '2.2.1', '2.2.2'];
+            throw new InvalidArgumentException(
+                sprintf(
+                    'QTI Result Report is not supported for version "%s". Supported versions are "%s".',
+                    $versionNumber,
+                    implode('", "', $knownVersions)
+                )
+            );
+        }
+
+        return $filename;
```

Comment:
Also, for me switch with immediate return, throwing exception at the bottom, would fit better here

Reply:
Totally agree. In the next PR, you'll see that. Even more, <spoiler>we finally change for an object support that let's each version to just provide the right filename</spoiler>

Reply:
I'm not convinced by putting a switch anywhere. Both are against the open-close principle. Perhaps this could be extracted to some kind of resolver?

Reply:
This is done in the next PR. We have a mechanism to identify which version the document is written for, and we then use a object representing this version, that provides all the filenames, xsd locations and all.